src/datagrid/DataGrid.tsx

```diff
@@ -516,18 +521,27 @@ export class DataGrid extends React.PureComponent<DataGridProps, DataGridState>
 
     private updateRowIDs(rows: DataGridRow[]) {
         // set rowID = index in array
-        rows.map((row: DataGridRow, index: number) => {
+        rows.forEach(function(row: DataGridRow, index: number) {
```

Comment:
this function is not pure. Use map instead. Here you modify the object that is a reason of changes of the incoming arg

Reply:
done